### ISSUE TITLE
Multi-intent classification

### ISSUE THREAD

**Issue by aminekechaou** [RasaHQ/rasa#6381]
The current way to handle multiple intents is to explicitly add the different combinations of intents in the training data. However, this is unscalable for a high number of intents and it is nearly impossible to enumerate all the combinations. My suggestion is to treat intent classification as a multi-way classification in a similar fashion to using sigmoid + binary cross entropy instead of softmax + cross entropy. This way intents are classified independently of one another. As a result, for example, if two intents get a score higher than 0.5, the utterance would be interpreted as having both intents. 

**EDIT:** It seems that setting the loss type to "margin" in the DIET classifier (see list of all parameters https://rasa.com/docs/rasa/nlu/components/#diet-classifier) achieves the intended purpose. Any thoughts on this approach to handle multiple intents?

**Comment by sara-tagger** [comment]
Thanks for submitting this feature request 🚀 @amn41 will get back to you about it soon! ✨

**Comment by amn41** [comment]
hi @aminekechaou ! there are a couple of reasons why we do things the way we do:
1. there are many possible combinations of intents, most of them will never occur in real life. So it's actually an advantage (in my opinion) to only predict combinations of intents that users have actually expressed. 

2. you would need your dialogue manager to handle each of these intent combinations that you've never seen, and it's not straightfoward to think about how handle them (e.g. can you ignore one of the intents, do you respond to both, do they conflict and you need to ask the user to clarify).
